service/internal/checksum/middleware_checksum_metrics_tracking.go

```diff
@@ -0,0 +1,90 @@
+package checksum
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+)
+
+var supportedChecksumFeatures = map[Algorithm]awsmiddleware.UserAgentFeature{
+	AlgorithmCRC32:     awsmiddleware.UserAgentFeatureRequestChecksumCRC32,
+	AlgorithmCRC32C:    awsmiddleware.UserAgentFeatureRequestChecksumCRC32C,
+	AlgorithmSHA1:      awsmiddleware.UserAgentFeatureRequestChecksumSHA1,
+	AlgorithmSHA256:    awsmiddleware.UserAgentFeatureRequestChecksumSHA256,
+	AlgorithmCRC64NVME: awsmiddleware.UserAgentFeatureRequestChecksumCRC64,
+}
+
+// RequestChecksumMetricsTracking is the middleware to track operation request's checksum usage
+type RequestChecksumMetricsTracking struct {
+	RequestChecksumCalculation aws.RequestChecksumCalculation
+	UserAgent                  *awsmiddleware.RequestUserAgent
+}
+
+// ID provides the middleware identifier
+func (m *RequestChecksumMetricsTracking) ID() string {
+	return "AWSChecksum:RequestMetricsTracking"
```

Comment:
[observation] we don't follow this pattern of `x:y`on other middlewares, but I like how you used it here